src/components/common/CollapseButton.js

```diff
@@ -0,0 +1,89 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import IconButton from '@material-ui/core/IconButton';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import ExpandLessIcon from '@material-ui/icons/ExpandLess';
+import { useTranslation } from 'react-i18next';
+import Tooltip from '@material-ui/core/Tooltip';
+import MenuItem from '@material-ui/core/MenuItem';
+import ListItemIcon from '@material-ui/core/ListItemIcon';
+import { MUTATION_KEYS } from '@graasp/query-client';
+import { useMutation } from '../../config/queryClient';
+import { COLLAPSE_ITEM_BUTTON_CLASS } from '../../config/selectors';
+import { BUTTON_TYPES } from '../../config/constants';
+
+const CollapseButton = ({ item, type, onClick }) => {
+  const { t } = useTranslation();
+
+  const editItem = useMutation(MUTATION_KEYS.EDIT_ITEM);
+  const [isCollapsible, setIsCollapsible] = useState(
+    item?.settings?.isCollapsible,
+  );
```

Comment:
I would add a useEffect in addition to be sure this value is sync with your state. That should mean that you don't need line 24.